pytorch_lightning/accelerators/gpu.py

```diff
@@ -38,3 +40,10 @@ def set_nvidia_flags():
         all_gpu_ids = ",".join([str(x) for x in range(torch.cuda.device_count())])
         devices = os.getenv("CUDA_VISIBLE_DEVICES", all_gpu_ids)
         _log.info(f"LOCAL_RANK: {os.getenv('LOCAL_RANK', 0)} - CUDA_VISIBLE_DEVICES: [{devices}]")
+
+    def to_device(self, batch: Any) -> Any:
```

Comment:
Actually I'm afraid we shouldn't add this right now, since that way the user cannot device custom batch-transfers for custom types. This is already true in the current case, but we should rather enable this instead of introducing more blockers.

The found solution can then be applied to this fix as well :)

Reply:
even if they have a custom batch, will DataParallel be able to split that batch across each gpu??

Reply:
That's what I meant :) Yeah. currently this is an issue for us. 
If we want to proceed with this, maybe just mention it in a comment and the docs?

Reply:
done.